lib/filter-headers.js

```diff
@@ -2,15 +2,17 @@
 const ACCEPT_HEADERS = ['accept-language', 'referer', 'user-agent'];
 /**
  * Filter the request headers that are passed to fragment request.
+ * @callback filterHeaders
  *
  * @param {Object} attributes - Attributes object of the fragment node
  * @param {string} attributes.public - Denotes the public fragemnt. Headers are not forward in this case
- * @param {Object} headers - Request header object
+ * @param {Object} request - HTTP Request object
+ * @param {Object} request.headers - request header object
  * @returns {Object} New filtered header object
  */
-module.exports = ({ public: publicFragment }, headers) => 
-    publicFragment 
-        ? {} 
+module.exports = ({ public: publicFragment }, { headers = {} }) =>
```

Comment:
Personal preference:

Since this is the `filterHeaders` function, I would expect the actual headers to filter as a first argument. With the given signature, we always need to extract the headers from the request object (either via argument matching or property access).

Alternative:

```
(headers, attributes, request) => {}
// or
(headers, context) => {} with { attributes, request } = context
```

So if you're not interested in the context, you can simply write:

```
(headers) => {}
```